tests/conftest.py

```diff
@@ -27,15 +27,177 @@
+    drop_database
+
+from zenodo_accessrequests import ZenodoAccessRequests
+from zenodo_accessrequests.views.requests import blueprint as request_blueprint
+from zenodo_accessrequests.views.settings import \
+    blueprint as settings_blueprint
 
 
 @pytest.fixture()
-def app():
+def app(request):
     """Flask application fixture."""
-    app = Flask('testapp')
-    app.config.update(
-        TESTING=True
+    app_ = Flask('testapp')
+    app_.config.update(
+        TESTING=True,
+        CELERY_ALWAYS_EAGER=True,
+        CELERY_CACHE_BACKEND="memory",
+        CELERY_EAGER_PROPAGATES_EXCEPTIONS=True,
+        CELERY_RESULT_BACKEND="cache",
+        SQLALCHEMY_DATABASE_URI=os.environ.get(
+            'SQLALCHEMY_DATABASE_URI', 'sqlite://'
+        ),
+        SECRET_KEY='mysecret',
+        SUPPORT_EMAIL='info@zenodo.org',
+        WTF_CSRF_ENABLED=False,
+        SERVER_NAME='test.it',
+        RECORDS_UI_ENDPOINTS=dict(
+            recid=dict(
+                pid_type='recid',
+                route='/records/<pid_value>',
+                template='invenio_records_ui/detail.html',
+            ),
+            recid_access_request=dict(
+                pid_type='recid',
+                route='/records/<pid_value>/accessrequest',
+                template='zenodo_accessrequests/access_request.html',
+                view_imp='zenodo_accessrequests.views.requests.access_request',
+                methods=['GET', 'POST'],
+            ),
+            recid_access_request_email_confirm=dict(
+                pid_type='recid',
+                route='/records/<pid_value>/accessrequest/<token>/confirm',
+                #  template='invenio_records_ui/detail.html',
+                view_imp='zenodo_accessrequests.views.requests.confirm',
+            ),
+        ),
     )
-    return app
+
+    Babel(app_)
+    InvenioDB(app_)
+    InvenioAccounts(app_)
+    InvenioRecords(app_)
+    FlaskMenu(app_)
+    Mail(app_)
+    InvenioRecordsUI(app_)
+    InvenioAccess(app_)
+    ZenodoAccessRequests(app_)
+    InvenioPIDStore(app_)
+
+    app_.register_blueprint(request_blueprint)
+    app_.register_blueprint(settings_blueprint)
+    app_.register_blueprint(blueprint_user)
+
+    return app_
+
+
+@pytest.fixture
+def db(app, request):
+    """Setup database."""
+    with app.app_context():
+        if not database_exists(str(db_.engine.url)):
+            create_database(str(db_.engine.url))
+        db_.create_all()
+
+    def teardown():
+        with app.app_context():
+            drop_database(str(db_.engine.url))
+            # Delete sessions in kvsession store
+            if hasattr(app, 'kvsession_store') and \
+                    isinstance(app.kvsession_store, RedisStore):
+                app.kvsession_store.redis.flushall()
+        #  shutil.rmtree(app.instance_path)
+
+    request.addfinalizer(teardown)
+    return db_
+
+
+@pytest.fixture
+def users(app, db):
+    """Demo users."""
+    with app.app_context():
```

Comment:
``with app.app_context():`` can be removed if two above changes are implemented. ditto for the remaining fixtures.